builtin/providers/aws/resource_aws_config_configuration_recorder.go

```diff
@@ -0,0 +1,145 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/configservice"
+)
+
+func resourceAwsConfigConfigurationRecorder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsConfigConfigurationRecorderPut,
+		Read:   resourceAwsConfigConfigurationRecorderRead,
+		Update: resourceAwsConfigConfigurationRecorderPut,
+		Delete: resourceAwsConfigConfigurationRecorderDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "default",
+				ValidateFunc: validateMaxLength(256),
+			},
+			"role_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"recording_group": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"all_supported": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						"include_global_resource_types": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"resource_types": {
+							Type:     schema.TypeSet,
```

Comment:
Because the ordering doesn't matter and because I wanted to be sure we don't generate diffs when AWS API decides to return types in a different order. :)